### DISCUSSION TITLE
How to add multiple variants of the same package for cross compilation?

### DISCUSSION THREAD

**Discussion by NickHu** [90-008/nix-cargo-integration#121] [Q&A] [answered]
I see there is an example for cross compiling for wasm: https://github.com/yusdacra/nix-cargo-integration/tree/master/examples/cross-compile-wasm

I am wondering if there is a simple way to make a flake with nci export multiple packages for the **same crate** but compiled for different targets.

**Comment by 90-008** [comment]
There is no API for this yet, but here is a draft API I had in mind:
```nix
{
	nci.crates."your-crate".targets."wasm32-unknown-unknown" = {
		# optionally choose which profiles to generate packages for this target
		profiles = ["dev" "release"];
		# extra features to enable for this target specifically
		features = ["some-target-specific-feature"];
		# extra derivation config specific to this target
		drvConfig = {};
		depsDrvConfig = {};
	};
}
```

**Reply by 90-008** [reply]
I merged a PR that implements this, and updated the cross-compile-wasm example to use it. I still need to add an actual example showcasing compilation into different targets but I think the current example should be enough to figure it out for now.

**Comment by 90-008** [comment, marked as answer]
there are now plenty of examples for cross compilation so this should not be an issue anymore
